### PR TITLE
Test: Added InspectMessageMethodArb

### LINKED ISSUE #1517: inspect_message
The canister may look something like this:

```ts
import { bool, Canister, ic, inspectMessage, Null, update, Variant, Void } from 'azle';

const InspectMessageOutcome = Variant({
    Accept: Null,
    Return: Null,
    Throw: Null
});
type InspectMessageOutcome = typeof InspectMessageOutcome.tsType;

let inspectMessageOutcome: InspectMessageOutcome = { Accept: null };

export default Canister({
    inspectMessage: inspectMessage(() => {
        if ('Accept' in inspectMessageOutcome) {
            ic.acceptMessage();
            return;
        }

        if ('Return' in InspectMessageOutcome) {
            return;
        }

        throw `Method "${ic.methodName()}" not allowed`;
    }),
    inspectMessageTester: update([], bool, () => {
        return true;
    }),
    setInspecteMessageTest: update([InspectMessageOutcome], Void, (outcome) => {
        inspectMessageOutcome = outcome;
    })
});
```

- [x] Tests accepting a message by calling `ic.acceptMessage()`
- [x] Tests rejecting a message by returning without calling `ic.acceptMessage`
- [x] Tests rejecting a message by throwing
- [x] Callback can either be defined inline or as a standalone function

> [!Note]
> `inspect_message` is not invoked for `query` methods

- [x] Test multiple behaviors each round

### PR DESCRIPTION
**Pull request by dansteren**
> [!Caution]
> Change the base of this brach to `main` before merging.

Closes #1517

### CODE REVIEW

**Review by lastmjs — CHANGES_REQUESTED** [review]
Let's manually play with it and make sure it's testing what we think it's testing, and more than one behavior per round of tests

File: dfx/index.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export function getAgentHost(): string {
         : `http://127.0.0.1:${getWebServerPort()}`;
 }
 
-export async function createAnonymousAgent() {
+export async function getAnonymousAgent() {
```
Comment (lastmjs) [inline dfx/index.ts:31]: Rename to createAnonymousAgent

File: dfx/index.ts [hunk 2]
```diff
@@ -38,7 +38,13 @@ export async function createAnonymousAgent() {
...
 }
 
-export async function createAuthenticatedAgent(
+/**
+ * Returns an agent authenticated with the identity with the given name.
+ *
+ * @param identityName
+ * @returns
+ */
+export async function getAuthenticatedAgent(
```
Comment (lastmjs) [inline dfx/index.ts:47]: Rename to createAuthenticatedAgent

File: dfx/index.ts [hunk 3]
```diff
@@ -53,6 +59,61 @@ export async function createAuthenticatedAgent(
...
+/**
+ * Returns an agent authenticated with the identity with the given name.
+ *
+ * IMPORTANT: In order to be synchronous this call will not fetch the root key.
+ * If you are not on mainnet you will need to fetch the root key separately.
+ *
+ * @param identityName
+ * @returns
+ */
+export function getAuthenticatedAgentSync(identityName: string): HttpAgent {
```
Comment (lastmjs) [inline dfx/index.ts:71]: Rename to createAuthenticatedAgentSync

File: dfx/index.ts [hunk 4]
```diff
@@ -53,6 +59,61 @@ export async function createAuthenticatedAgent(
...
+ * an identity with that name, and returns an agent authenticated with that
+ * identity.
+ *
+ * IMPORTANT: In order to be synchronous this call will not fetch the root key.
+ * If you are not on mainnet you will need to fetch the root key separately.
+ *
+ * @param identityName
+ * @returns
+ */
+export function generateAuthenticatedAgentSync(
```
Comment (lastmjs) [inline dfx/index.ts:91]: Collapse all down into one function with a default parameter for creating the identity if it does not exist

File: dfx/index.ts [hunk 5]
```diff
@@ -53,6 +59,61 @@ export async function createAuthenticatedAgent(
...
+
+/**
+ * Generates a new identity with the given name if there doesn't already exist
+ * an identity with that name, and returns an agent authenticated with that
+ * identity.
+ *
+ * @param identityName
+ * @returns
+ */
+export async function generateAuthenticatedAgent(
```
Comment (lastmjs) [inline dfx/index.ts:108]: Collapse all down into one function with a default parameter for creating the identity if it does not exist

File: property_tests/get_actor.ts [hunk 6]
```diff
@@ -13,10 +15,16 @@ export function getActor(parentDir: string) {
...
     // eslint-disable-next-line @typescript-eslint/no-var-requires
     const { createActor } = require(`${parentDir}/dfx_generated/canister`);
-    return createActor(getCanisterId('canister'), {
-        agentOptions: {
-            host: 'http://127.0.0.1:8000',
-            verifyQuerySignatures: false // TODO Major issue: https://forum.dfinity.org/t/agent-js-0-20-0-is-released-replica-signed-query-edition/24743/16?u=lastmjs
-        }
-    });
+
+    const options =
```
Comment (lastmjs) [inline property_tests/get_actor.ts:19]: Clean up

File: property_tests/tests/canister_methods/inspect_message/test/generate_tests.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+import { CorrespondingJSType } from 'azle/property_tests/arbitraries/candid/corresponding_js_type';
+import { Test } from 'azle/test';
+
+import { InspectMessageBehavior } from './test';
+
+export function generateTests(
+    functionName: string,
+    params: Named<CandidValueAndMeta<CorrespondingJSType>>[],
+    returnType: CandidValueAndMeta<CandidReturnType>,
+    agents: [Agent, InspectMessageBehavior][]
```
Comment (lastmjs) [inline property_tests/tests/canister_methods/inspect_message/test/generate_tests.ts:14]: Rename agents...agentInfo?

Comment (lastmjs) [reply property_tests/tests/canister_methods/inspect_message/test/generate_tests.ts:14]: agentAndBehaviors

File: property_tests/tests/canister_methods/inspect_message/test/test.ts [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    const acceptPrincipal = getPrincipal(AZLE_ACCEPT_IDENTITY_NAME);
+    const returnPrincipal = getPrincipal(AZLE_RETURN_IDENTITY_NAME);
+    const throwPrincipal = getPrincipal(AZLE_THROW_IDENTITY_NAME);
+    return `
+        if (ic.caller().toText() === "${acceptPrincipal}") {
+            ic.acceptMessage()
+            return;
+        }
+        if (ic.caller().toText() === "${returnPrincipal}") {
+            return
```
Comment (lastmjs) [inline property_tests/tests/canister_methods/inspect_message/test/test.ts:78]: Semicolons

File: property_tests/arbitraries/canister_methods/inspect_message_method_arb.ts [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    generateTests: TestsGenerator;
+    callbackLocation?: CallbackLocation;
+}) {
+    return fc
+        .tuple(
+            UniqueIdentifierArb('canisterMethod'),
+            VoidArb(),
+            CallbackLocationArb,
+            UniqueIdentifierArb('typeDeclaration')
+            // TODO: This unique id would be better named globalScope or something
```
Comment (lastmjs) [inline property_tests/arbitraries/canister_methods/inspect_message_method_arb.ts:32]: Look into this todo

Comment (lastmjs) [reply property_tests/arbitraries/canister_methods/inspect_message_method_arb.ts:32]: Change uniqueidentifierarb parameters

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: dfx/index.ts (hunk 1 above)
Comment (bdemann) [reply dfx/index.ts:31]: Done

File: dfx/index.ts (hunk 2 above)
Comment (bdemann) [reply dfx/index.ts:47]: Done

File: dfx/index.ts (hunk 3 above)
Comment (bdemann) [reply dfx/index.ts:71]: Done

File: dfx/index.ts (hunk 4 above)
Comment (bdemann) [reply dfx/index.ts:91]: Done

File: dfx/index.ts (hunk 5 above)
Comment (bdemann) [reply dfx/index.ts:108]: Done

File: property_tests/tests/canister_methods/inspect_message/test/test.ts (hunk 8 above)
Comment (bdemann) [reply property_tests/tests/canister_methods/inspect_message/test/test.ts:78]: done

File: property_tests/tests/canister_methods/inspect_message/test/generate_tests.ts (hunk 7 above)
Comment (bdemann) [reply property_tests/tests/canister_methods/inspect_message/test/generate_tests.ts:14]: done

File: property_tests/arbitraries/canister_methods/inspect_message_method_arb.ts (hunk 9 above)
Comment (bdemann) [reply property_tests/arbitraries/canister_methods/inspect_message_method_arb.ts:32]: Done

File: property_tests/get_actor.ts (hunk 6 above)
Comment (bdemann) [reply property_tests/get_actor.ts:19]: Done

**Review by lastmjs — DISMISSED** [review]

**Review by lastmjs — APPROVED** [review]

**Pull request merged**